src/main/java/codeu/controller/ChatServlet.java

```diff
@@ -220,15 +220,28 @@ public void doPut(HttpServletRequest request, HttpServletResponse response)
       return;
     }
 
-    String purpose = request.getReader().readLine();
+    String purpose = request.getHeader("purpose");
+    if(purpose == null){
```

Comment:
Nice job using the header to determine the needed action, looks clean and it's easy to understand. 👍